src/iuse_actor.cpp

```diff
@@ -277,14 +278,15 @@ std::optional<int> iuse_transform::use( Character *p, item &it, const tripoint &
     return result;
 }
 
-void iuse_transform::do_transform( Character *p, item &it ) const
+void iuse_transform::do_transform( Character *p, item &it, const std::string variant_type ) const
```

Comment:
You need the & or else it copies.
```suggestion
void iuse_transform::do_transform( Character *p, item &it, const std::string &variant_type ) const
```